cmd/remote_enforcer.go

```diff
@@ -0,0 +1,291 @@
+	"github.com/aporeto-inc/trireme/cache"
+	"github.com/aporeto-inc/trireme/collector"
+	"github.com/aporeto-inc/trireme/enforcer"
+	_ "github.com/aporeto-inc/trireme/enforcer/utils/nsenter"
+	"github.com/aporeto-inc/trireme/enforcer/utils/packet"
+	"github.com/aporeto-inc/trireme/enforcer/utils/rpcwrapper"
+	"github.com/aporeto-inc/trireme/enforcer/utils/tokens"
+	"github.com/aporeto-inc/trireme/policy"
+	"github.com/aporeto-inc/trireme/supervisor"
+	"github.com/aporeto-inc/trireme/supervisor/iptablesutils"
+	"github.com/aporeto-inc/trireme/supervisor/provider"
+)
+
+const (
+	ipcProtocol         = "unix"
+	defaultPath         = "/var/run/default.sock"
+	statsContextID      = "UNUSED"
+	defaulttimeInterval = 2
+	envStatsChannelPath = "STATSCHANNEL_PATH"
+	envSocketPath       = "SOCKET_PATH"
+)
+
+//CollectorImpl exported
+type CollectorImpl struct {
+	Flowentries *list.List
+	sync.RWMutex
+}
+type collectorentry struct {
+	L4FlowHash string
+	entry      *enforcer.StatsPayload
+}
+
+//CollectFlowEvent expoted
+func (c *CollectorImpl) CollectFlowEvent(contextID string, tags policy.TagsMap, action string, mode string, sourceID string, tcpPacket *packet.Packet) {
+	l4FlowHash := tcpPacket.L4FlowHash()
+	payload := &enforcer.StatsPayload{ContextID: contextID,
+		Tags:   tags,
+		Action: action,
+		Mode:   mode,
+		Source: sourceID,
+		Packet: tcpPacket,
+	}
+
+	c.Lock()
+	c.Flowentries.PushBack(&collectorentry{L4FlowHash: l4FlowHash, entry: payload})
+	c.Unlock()
+}
+
+//CollectContainerEvent exported
+//This event should not be expected here in the enforcer process inside a particular container context
+func (c *CollectorImpl) CollectContainerEvent(contextID string, ip string, tags policy.TagsMap, event string) {
+}
+
+//Server exported
+type Server struct {
+	MutualAuth  bool
+	Validity    time.Duration
+	SecretType  tokens.SecretsType
+	ContextID   string
+	CAPEM       []byte
+	PublicPEM   []byte
+	PrivatePEM  []byte
+	rpcchannel  string
+	rpchdl      *rpcwrapper.RPCWrapper
+	StatsClient *rpcwrapper.RPCWrapper
+	Enforcer    enforcer.PolicyEnforcer
+	Collector   collector.EventCollector
+	Supervisor  supervisor.Supervisor
+}
+
+type StatsClient struct {
+	collector *CollectorImpl
+	server    *Server
+	FlowCache *cache.Cache
+	Rpchdl    *rpcwrapper.RPCWrapper
+}
+
+func (s *StatsClient) SendStats() {
+	//We are connected and lets pack and ship
+	rpcpayload := &rpcwrapper.StatsPayload{}
+	var request rpcwrapper.Request
+	var response rpcwrapper.Response
+	var statsInterval time.Duration
+	rpcpayload.NumFlows = 0
+	EnvstatsInterval, err := strconv.Atoi(os.Getenv("STATS_INTERVAL"))
+
+	if err == nil && EnvstatsInterval != 0 {
+		statsInterval = time.Duration(EnvstatsInterval) * time.Second
+	} else {
+		statsInterval = defaulttimeInterval * time.Second
+	}
+
+	starttime := time.Now()
+	for {
+		s.collector.Lock()
+		for !(s.collector.Flowentries.Len() > 0) {
+			s.collector.Unlock()
+			time.Sleep(statsInterval)
+			s.collector.Lock()
+			continue
+		}
```

Comment:
This loop could be replaced b a if statement I guess
```
if s.collector.Flowentries.Len() == 0 {
  s.collector.Unlock()
  time.Sleep(statsInterval)
} else {
....
```

Flowentries should be FlowEntries probably